### PR TITLE
Added groovy script and API changes based on model change

### PR DESCRIPTION
**Pull request by rajusem**
- Added groovy script and related logic to run it at startup
- Added file to insert bulk data as well as do load test for few gremlin query
- Modified ingest historical logic
- Addded few unit test cases

### CODE REVIEW

**Comment by rajusem** [comment]
[test]

File: src/rest_model.py [hunk 1]
```diff
@@ -20,65 +20,92 @@
...
 class ISO8601Format(fields.Raw):
     """Abstracts iso8601 string formatting."""
 
+    __schema_type__ = 'string'
+    __schema_example__ = '2020-02-23 19:02:15+00:00'
+
     def format(self, value):
         """Convert given epoch to iso8601 string."""
-        return to_date_str(value)
+        return None if value is None else to_date_str(value)
```
Comment (rootAvish) [inline src/rest_model.py:45]: `return to_date_str(value) if value else None` is more clear.

File: src/ingestion_data.py [hunk 2]
```diff
@@ -15,24 +15,16 @@ def __init__(self, json_data):
...
-        except:
-            return None
+        return from_date_str(self._payload[field_name])
 
     def _updated_at(self) -> int:
         return self._timestamp('updated_at')
 
     def _closed_at(self) -> int:
-        return self._timestamp('closed_at')
+        return self._timestamp('closed_at') if self._payload['closed_at'] is not None else None
```
[the remark this answers is not in the document]
Comment (rajusem) [reply src/ingestion_data.py:27]: we are calling `self._timestamp('closed_at')` if `closed_at` in payload else just returning `None`. So I guess code you suggested won't work here.

File: populate_schema.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        return False, json_response
+    else:
+        return True, json_response
+
+
+def populate_schema():
+    """Populate the schema stored in the Groovy script."""
+    current_file_path = os.path.dirname(os.path.realpath(__file__))
+    schema_file_path = os.path.join(current_file_path, 'scripts/schema.groovy')
+    str_gremlin_dsl = ''''''
```
[the remark this answers is not in the document]
Comment (rajusem) [reply populate_schema.py]: deleted this line as not required now.

File: populate_schema.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    with open(schema_file_path, 'r') as f:
+        str_gremlin_dsl = f.read()
+
+    return execute(str_gremlin_dsl)
```
[the remark this answers is not in the document]
Comment (rajusem) [reply populate_schema.py]: done

File: src/rest_model.py (hunk 1 above)
Comment (rajusem) [reply src/rest_model.py]: done

File: populate_schema.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    """Execute the query prepared for the graph database."""
+    logger.info("Execute gremlin DSL : {dsl}".format(dsl=str_gremlin_dsl))
+    payload = {'gremlin': str_gremlin_dsl}
+    response = requests.post(config.GREMLIN_URL, data=json.dumps(payload))
+    json_response = response.json()
+
+    logger.info("Execute Response : {res}".format(res=response))
+    if response.status_code != 200:
+        logger.error("ERROR with code: {code}, reason: {reason}, msg: {msg}".format(code=str(response.status_code),
+                                                                                    reason=response.reason,
```
[the remark this answers is not in the document]
Comment (rajusem) [reply populate_schema.py]: There were no formatting issue but might be little lengthy code was there, refactored it.

**Comment by rajusem** [comment]
> Some minor code style comments, otherwise looks good.

@rootAvish : Incoreporated suggestions given by you, please have a look.

**Pull request merged**